trace-mini-agent/src/mini_agent.rs

```diff
@@ -51,11 +51,24 @@ impl MiniAgent {
         });
 
         let addr = SocketAddr::from(([127, 0, 0, 1], MINI_AGENT_PORT as u16));
-        let server = Server::bind(&addr).serve(make_svc);
+        let server_builder = match Server::try_bind(&addr) {
```

Comment:
you could use `map_err` to simplify this code block

Reply:
simplified this code block to propagate error back to whatever called start_mini_agent